packages/store/src/StoreCore.sol

```diff
@@ -0,0 +1,559 @@
+    // Get schema for this table
+    Schema schema = getSchema(table);
+
+    // Call onDeleteRecord hooks (before actually modifying the state, so observers have access to the previous state if needed)
+    address[] memory hooks = HooksTable.get(table);
+    for (uint256 i = 0; i < hooks.length; i++) {
+      IStoreHook hook = IStoreHook(hooks[i]);
+      hook.onDeleteRecord(table, key);
+    }
+
+    // Delete static data
+    uint256 staticDataLocation = StoreCoreInternal._getStaticDataLocation(table, key);
+    Storage.store({ storagePointer: staticDataLocation, offset: 0, data: new bytes(schema.staticDataLength()) });
+
+    // If there are no dynamic fields, we're done
+    if (schema.numDynamicFields() == 0) return;
+
+    // Delete dynamic data
+    uint256 dynamicDataLocation;
+    PackedCounter encodedLengths = StoreCoreInternal._loadEncodedDynamicDataLength(table, key);
+    for (uint8 i; i < schema.numDynamicFields(); ) {
+      dynamicDataLocation = StoreCoreInternal._getDynamicDataLocation(table, key, i);
+      Storage.store({ storagePointer: dynamicDataLocation, offset: 0, data: new bytes(encodedLengths.atIndex(i)) });
+      unchecked {
+        i++;
+      }
+    }
+
+    // Delete dynamic data length
+    uint256 dynamicDataLengthLocation = StoreCoreInternal._getDynamicDataLengthLocation(table, key);
+    Storage.store({ storagePointer: dynamicDataLengthLocation, data: bytes32(0) });
+
+    // Emit event to notify indexers
+    emit MudStoreDeleteRecord(table, key);
+  }
+
+  /************************************************************************
+   *
+   *    GET DATA
+   *
+   ************************************************************************/
+
+  /**
+   * Get full static record for the given table and key tuple (loading schema from storage)
+   */
+  function getRecord(bytes32 table, bytes32[] memory key) internal view returns (bytes memory) {
+    // Get schema for this table
+    Schema schema = getSchema(table);
+    if (schema.isEmpty()) revert StoreCore_TableNotFound(table);
+
+    return getRecord(table, key, schema);
+  }
+
+  /**
+   * Get full data (static and dynamic) for the given table and key tuple, with the given schema
+   */
+  function getRecord(
+    bytes32 table,
+    bytes32[] memory key,
+    Schema schema
+  ) internal view returns (bytes memory) {
+    // Get the static data length
+    uint256 outputLength = schema.staticDataLength();
+
+    // Load the dynamic data length if there are dynamic fields
+    PackedCounter dynamicDataLength;
+    uint256 numDynamicFields = schema.numDynamicFields();
+    if (numDynamicFields > 0) {
+      dynamicDataLength = StoreCoreInternal._loadEncodedDynamicDataLength(table, key);
+      outputLength += 32 + dynamicDataLength.total(); // encoded length + data
+    }
+
+    // Allocate a buffer for the full data (static and dynamic)
+    Buffer buffer = Buffer_.allocate(uint128(outputLength));
```

Comment:
Coming back to [the Buffer comment](https://github.com/latticexyz/mud/pull/352/files#r1096575924) - here it could be something like:
```solidity
bytes memory result = new bytes(outputLength);
Buffer buffer = Buffer_.fromBytes(result);
```
I guess I'm suggesting a `Slice` at this point, not `Buffer`

Reply:
Agree with this!